### PR TITLE
Refactor mapping list

### PR DESCRIPTION
**Pull request by ghost**
Simplification of mapping list CSS and component code, with added comments for clarity.

### CODE REVIEW

File: ui/admin/app/components/mapping-list/index.hbs [hunk 1]
```diff
@@ -36,20 +38,21 @@
                 {{on 'input' (set-from-event option 'value')}}
                 as |F|
               >
-                <F.Label>{{t 'form.value.label'}}</F.Label>
+                <F.Label classes='sr-only'>{{t 'form.value.label'}}</F.Label>
```
Comment (randallmorey) [inline ui/admin/app/components/mapping-list/index.hbs:41]:
```suggestion
                <F.Label class='sr-only'>{{t 'form.value.label'}}</F.Label>
```

File: ui/admin/app/components/mapping-list/index.hbs [hunk 2]
```diff
@@ -36,20 +38,21 @@
...
-                @text={{t 'actions.remove'}}
-                @color='critical'
-                @icon='trash'
-                @isIconOnly={{true}}
-                disabled={{@disabled}}
-                {{on 'click' (fn this.removeOptionByIndex index)}}
-              />
-
-            </B.Td>
+            {{#if @removeOptionByIndex}}
```
Comment (ZedLi) [inline ui/admin/app/components/mapping-list/index.hbs:44]: Should the display of these cells be controlled by whether we have the function? The reason why I had the indirect function call was to guard against not including the function but I think it should still be required.

Wouldn't this also not work with the `last-child` css that shrinks the last cell?

**Review by ZedLi — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/mapping-list/index.js [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
+   */
   @action
   addOption() {
-    if (this.args.addOption) {
+    assert(
+      '[boundary-admin-key-value-list-field] `@addOption` is required.',
+      this.args.addOption
+    );
+
+    if (this.newOptionKey) {
```
Comment (ZedLi) [inline ui/admin/app/components/mapping-list/index.js:32]: Do we want to always require a key to be able to add? If so, shouldn't we be disabling the add button instead?

File: ui/admin/app/styles/app.scss [hunk 4]
```diff
@@ -710,33 +710,61 @@
...
-    display: none;
-  }
+      &:last-child {
+        // ...except in the last cell, which doesn't need right padding
+        padding-right: 0;
 
-  border: 0;
+        // Shrink last cell to minimum possible width
+        width: 1%;
+        white-space: nowrap;
```
Comment (ZedLi) [inline ui/admin/app/styles/app.scss:743]: What's the purpose of adding `nowrap` here, shouldn't we always be expecting a button of some sort?

File: ui/admin/app/components/mapping-list/index.hbs (hunk 2 above)
Comment (randallmorey) [reply ui/admin/app/components/mapping-list/index.hbs:44]: Good point.

File: ui/admin/app/styles/app.scss (hunk 4 above)
Comment (randallmorey) [reply ui/admin/app/styles/app.scss:743]: In case the button has multiple words in it.

File: ui/admin/app/components/mapping-list/index.hbs (hunk 2 above)
Comment (ZedLi) [reply ui/admin/app/components/mapping-list/index.hbs]: Good to know, wasn't aware of assert usage

**Review by ZedLi — APPROVED** [review]
LGTM

**Pull request merged**